client/utils/reduxFormUtils.ts

```diff
@@ -0,0 +1,156 @@
+import i18n from 'i18next';
+
+/* eslint-disable */
+const EMAIL_REGEX = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/i;
+const USERNAME_REGEX = /^[a-zA-Z0-9._-]{1,20}$/
+/* eslint-enable */
+
+type Email = { email: string };
+type Username = { username: string };
+type Password = { password: string };
+type ConfirmPassword = { confirmPassword: string };
+type CurrentPassword = { currentPassword: string };
+type NewPassword = { newPassword: string };
+
+type UsernameAndEmail = Username & Email;
+type PasswordsConfirm = Password & ConfirmPassword;
+
+/** Validation errors for site forms */
+export type FormErrors = Partial<
+  Email & Username & Password & ConfirmPassword & CurrentPassword & NewPassword
+>;
+
+// === Internal helper functions: =====
+
+/** Processes form & mutates errors to add any `username` & `email` errors */
+function validateUsernameEmail(
+  formProps: Partial<UsernameAndEmail>,
+  errors: Partial<FormErrors>
```

Comment:
I'm not sure i understand what happens when we do Partial<Partial<....>> do we need the double partial?